src/renderer/routes/test_sequencer_panel/components/ControlButton.tsx

```diff
@@ -0,0 +1,96 @@
+import LockableButton from "./lockable/LockedButtons";
+import { useTestSequencerState } from "@/renderer/hooks/useTestSequencerState";
+import _ from "lodash";
+import {
+  testSequenceStopRequest,
+  testSequencePauseRequest,
+  testSequenceResumeRequest,
+} from "@/renderer/routes/test_sequencer_panel/models/models";
+import { TSWebSocketContext } from "@/renderer/context/testSequencerWS.context";
+import { useContext } from "react";
+import { Button } from "@/renderer/components/ui/button";
+import { Pause, Play, Square } from "lucide-react";
+import { toast } from "sonner";
+
+export function ControlButton() {
+  const { tree, setIsLocked, backendGlobalState, backendState, runSequencer } =
+    useTestSequencerState();
+  const { tSSendJsonMessage } = useContext(TSWebSocketContext);
+
+  const handleClickRunTest = () => {
+    console.log("Run sequencer");
+    runSequencer(tSSendJsonMessage);
+  };
+
+  const handleClickStopTest = () => {
+    toast.warning("Stopping sequencer after this test.");
+    tSSendJsonMessage(testSequenceStopRequest(tree));
+    setIsLocked(false);
+  };
+
+  const handleClickPauseTest = () => {
+    toast.warning("Pausing sequencer after this test.");
+    if (backendGlobalState === "test_set_start") {
+      console.log("Pause test");
+      tSSendJsonMessage(testSequencePauseRequest(tree));
+    }
+  };
+
+  const handleClickResumeTest = () => {
+    toast.info("Resuming sequencer.");
+    if (backendGlobalState === "test_set_start") {
+      console.log("Resume test");
+      tSSendJsonMessage(testSequenceResumeRequest(tree));
+    }
+  };
```

Comment:
these are probably things that should be exposed in a hook defined in the sequencer store file

Reply:
I'm migrating them to the sequencerState, so it's at the same place of the `runSequencer`